### PR TITLE
Add Jenkins XML reporting to tests

### PR DESCRIPTION
**Pull request by murgatroid99**
The XML file names are going to be based on timestamp and not anything useful, but it's better than nothing.

### CODE REVIEW

**Comment by murgatroid99** [comment]
I redid this change with the new script

File: run-tests.sh [hunk 1]
```diff
@@ -46,7 +46,8 @@ do
   ./node_modules/.bin/gulp setup
 
   # Rebuild libraries and run tests.
-  ./node_modules/.bin/gulp native.test || FAILED="true"
+  JUNIT_REPORT_PATH="node$version/" JUNIT_REPORT_STACK=1 ./node_modules/.bin/gulp native.test || FAILED="true"
+  rename 's/([^\/]+).xml/$1_sponge_log.xml/' "node$version"/*
```
Comment (nicolasnoble) [inline run-tests.sh]: ./run-tests.sh: line 50: rename: command not found

Comment (nicolasnoble) [reply run-tests.sh]: Suggest

```
cd node$version
for file in * ; do
  mv $f `echo $f | sed 's/\(.*\)\.xml/\1_sponge_log.xml/'`
fi
```

(also `.xml` in the regex is wrong, it should be `\.xml` :-) )

**Comment by matt-kwong** [comment]
It looks like Sponge has a weird way of parsing the .xml files. I think it interprets multiple .xml in the same directory as reruns of a flaky test. Can this be changed so that each .xml file has its own subdirectory or that the .xml files are combined?

**Comment by matt-kwong** [comment]
You can see the current directory structure of the reports generated here: https://pantheon.corp.google.com/storage/browser/grpc-testing-kokoro-prod/test_result_public/prod/grpc/node/pull_request/linux/220/20170919-142854/github/grpc-node/reports/

**Pull request merged**